streams/shared/src/main/scala/zio/stream/ZTransducer.scala

```diff
@@ -32,6 +32,75 @@ abstract class ZTransducer[-R, +E, -I, +O](
       }
     }
 
+  /**
+   * Symbolic alias for [[ZTransducer#zip]].
+   */
+  final def <&>[R1 <: R, E1 >: E, I1 <: I, O2](that: ZTransducer[R1, E1, I1, O2]): ZTransducer[R1, E1, I1, (O, O2)] =
+    self zip that
+
+  /**
+   * Symbolic alias for [[ZTransducer#zipLeft]].
+   */
+  final def <&[R1 <: R, E1 >: E, I1 <: I, O2](that: ZTransducer[R1, E1, I1, O2]): ZTransducer[R1, E1, I1, O] =
+    self zipLeft that
+
+  /**
+   * Symbolic alias for [[ZTransducer#zipRight]].
+   */
+  final def &>[R1 <: R, E1 >: E, I1 <: I, O2](that: ZTransducer[R1, E1, I1, O2]): ZTransducer[R1, E1, I1, O2] =
+    self zipRight that
+
+  /**
+   * Zips this transducer with another point-wise, but keeps only the outputs of this transducer.
+   */
+  def zipLeft[R1 <: R, E1 >: E, I1 <: I, O2](that: ZTransducer[R1, E1, I1, O2]): ZTransducer[R1, E1, I1, O] =
+    zipWith(that)((o, _) => o)
+
+  /**
+   * Zips this transducer with another point-wise, but keeps only the outputs of the other transducer.
+   */
+  def zipRight[R1 <: R, E1 >: E, I1 <: I, O2](that: ZTransducer[R1, E1, I1, O2]): ZTransducer[R1, E1, I1, O2] =
+    zipWith(that)((_, o2) => o2)
+
+  /**
+   * Zips this transducer with another point-wise and emits tuples of elements from both transducers.
+   */
+  def zip[R1 <: R, E1 >: E, I1 <: I, O2](that: ZTransducer[R1, E1, I1, O2]): ZTransducer[R1, E1, I1, (O, O2)] =
+    zipWith(that)((_, _))
+
+  /**
+   * Zips this transducer with another point-wise and applies the function to the paired elements.
+   */
+  def zipWith[R1 <: R, E1 >: E, I1 <: I, O2, O3](
+    that: ZTransducer[R1, E1, I1, O2]
+  )(f: (O, O2) => O3): ZTransducer[R1, E1, I1, O3] = {
+    type State = Either[Chunk[O], Chunk[O2]]
+    ZTransducer {
+      for {
+        ref <- ZRef.make[State](Left(Chunk.empty)).toManaged_
+        p1  <- self.push
+        p2  <- that.push
+        push = (in: Option[Chunk[I1]]) => {
+          ref.get.flatMap { excess =>
+            for {
+              res <- p1(in).zipWithPar(p2(in)) {
```

Comment:
With this version it is possible that one slow stream will block a fast stream, as it should, but chunks will accumulate in memory potentially until collapse. I think it's worth pulling from one stream only when there's an excess, perhaps with a heuristic on the size of the excess.

Reply:
Hmm, no, I misread. Ha, I take it back 😅